ingestion/src/metadata/profiler/interface/sqlalchemy/profiler_interface.py

```diff
@@ -522,15 +522,23 @@ def fetch_sample_data(self, table) -> TableData:
         Returns:
             TableData: sample table data
         """
-        sampler = Sampler(
-            session=self.session,
-            table=table,
-            sample_columns=self._get_sample_columns(),
-            profile_sample_config=self.profile_sample_config,
-            partition_details=self.partition_details,
-            profile_sample_query=self.profile_query,
-        )
-
+        if self.service_connection_config.type.value == "BigQuery":
```

Comment:
instead of having if else here, it would be better to have overriding capabilities with-in each connector